src/libbson/src/bson/bson-string.c

```diff
@@ -159,24 +198,48 @@ bson_string_free (bson_string_t *string, /* IN */
  */
 
 void
-bson_string_append (bson_string_t *string, /* IN */
-                    const char *str)       /* IN */
+bson_string_append_ex (bson_string_t *string, /* IN */
+                       const char *str,       /* IN */
+                       const size_t len)      /* IN */
 {
    BSON_ASSERT (string);
    BSON_ASSERT (str);
 
-   const size_t len_sz = strlen (str);
-   BSON_ASSERT (bson_in_range_unsigned (uint32_t, len_sz));
-   const uint32_t len_u32 = (uint32_t) len_sz;
+   BSON_ASSERT (bson_in_range_unsigned (uint32_t, len));
+   const uint32_t len_u32 = (uint32_t) len;
    BSON_ASSERT (len_u32 <= UINT32_MAX - string->len);
    const uint32_t new_len = len_u32 + string->len;
    bson_string_ensure_space (string, new_len);
-   memcpy (string->str + string->len, str, len_sz);
+   memcpy (string->str + string->len, str, len);
    string->str[new_len] = '\0';
    string->len = new_len;
 }
 
 
+/*
+ *--------------------------------------------------------------------------
+ *
+ * bson_string_append --
+ *
+ *       Append the UTF-8 string @str to @string.
+ *
+ * Returns:
+ *       None.
+ *
+ * Side effects:
+ *       None.
+ *
+ *--------------------------------------------------------------------------
+ */
+
+void
+bson_string_append (bson_string_t *string, /* IN */
+                    const char *str)       /* IN */
+{
+   bson_string_append_ex (string, str, strlen (str));
```

Comment:
```suggestion
   BSON_ASSERT_PARAM (string);
   BSON_ASSERT_PARAM (str);
   bson_string_append_ex (string, str, strlen (str));
```
Assert preconditions + guard call to `strlen` (null argument is UB).